01_install_requirements.sh

```diff
@@ -24,65 +35,133 @@ fi
+    if [ ! -d crudini ] ; then
+        git clone https://github.com/pixelb/crudini
+    fi
+    pushd crudini
+    git pull -r
+    sudo pip3 install .
+    popd ; popd
+else
+    sudo yum -y install \
+        crudini \
+        python-pip \
+        python-requests \
+        python-setuptools
+fi
 sudo yum -y install \
-  crudini \
   curl \
   dnsmasq \
   golang \
   NetworkManager \
   nmap \
   patch \
   psmisc \
-  python-pip \
-  python-requests \
-  python-setuptools \
   vim-enhanced \
   wget
 
-# We're reusing some tripleo pieces for this setup so clone them here
-cd
-if [ ! -d tripleo-repos ]; then
-  git clone https://git.openstack.org/openstack/tripleo-repos
-fi
-pushd tripleo-repos
-sudo python setup.py install
-popd
+if [ "${RHEL8}" = "y" ] ; then
+    sudo subscription-manager repos --enable=ansible-2-for-rhel-8-x86_64-rpms
 
-# Needed to get a recent python-virtualbmc package
-sudo tripleo-repos current-tripleo
+    # make sure additional requirments are installed
+    sudo yum -y install \
+      ansible \
+      python3-netaddr \
+      redhat-lsb-core \
+      bind-utils \
+      jq \
+      libguestfs-tools \
+      libvirt \
+      libvirt-devel \
+      libvirt-daemon-kvm \
+      podman \
+      qemu-kvm \
+      virt-install \
+      unzip \
+      network-scripts \
+      ipmitool
 
-# There are some packages which are newer in the tripleo repos
-sudo yum -y update
+    sudo pip3 install yq
 
-# Setup yarn and nodejs repositories
-sudo curl -sL https://dl.yarnpkg.com/rpm/yarn.repo -o /etc/yum.repos.d/yarn.repo
-curl -sL https://rpm.nodesource.com/setup_10.x | sudo bash -
+    # TODO(russellb) - Install an rpm for this once OSP for RHEL8 is out
+    pushd ~
+    if [ ! -d virtualbmc ] ; then
+        git clone https://git.openstack.org/openstack/virtualbmc
+    fi
+    pushd virtualbmc
+    git pull -r
+    sudo pip3 install .
+    curl 'https://review.rdoproject.org/r/gitweb?p=openstack/virtualbmc-distgit.git;a=blob_plain;f=virtualbmc.service;hb=HEAD' > virtualbmc.service
+    sudo mv virtualbmc.service /etc/systemd/system/.
+    sudo systemctl daemon-reload
+    popd ; popd
 
-# make sure additional requirments are installed
-sudo yum -y install \
-  ansible \
-  bind-utils \
-  jq \
-  libguestfs-tools \
-  libvirt \
-  libvirt-devel \
-  libvirt-daemon-kvm \
-  nodejs \
-  podman \
-  python-ironicclient \
-  python-ironic-inspector-client \
-  python-lxml \
-  python-netaddr \
-  python-openstackclient \
-  python-virtualbmc \
-  qemu-kvm \
-  virt-install \
-  unzip \
-  yarn
-
-# Install python packages not included as rpms
-sudo pip install \
-  yq
+    # TODO(russellb) - Install an rpm for this once OSP for RHEL8 is out
+    sudo dnf groupinstall -y "Development Tools"
+    sudo dnf install -y python36-devel
+    pushd ~
+    if [ ! -d openstackclient ] ; then
+        git clone https://git.openstack.org/openstack/openstackclient
+    fi
+    pushd openstackclient
+    git pull -r
+    sudo pip3 install .
```

Comment:
This is going to pull in a lot of packages, randomly overriding system ones. Even for a temporary hack, I think we should use `--user` or a virtual env. By the way, if you want to have ironic commands, you have to install `ironicclient` and `ironic-inspector-client` as well.

Reply:
The ironic libraries seemed to have been installed automatically.  I have them on my dev box, anyway, and have only been installing things via these scripts.